src/frontend/src/pages/DRCPage/MonthSelector.tsx

```diff
@@ -0,0 +1,62 @@
+import { Box, MenuItem, TextField } from '@mui/material';
+import { Dispatch, SetStateAction } from 'react';
+
+interface MonthSelectorProps {
+  displayMonth: Date;
+  setDisplayMonth: Dispatch<SetStateAction<Date>>;
+}
+
+const MonthSelector: React.FC<MonthSelectorProps> = ({ displayMonth, setDisplayMonth }) => {
+  const monthNames = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'
+  ];
+  return (
+    <Box>
+      <TextField
+        select
+        defaultValue={monthNames[displayMonth.getMonth()]}
+        onChange={(event) => {
+          displayMonth.setMonth(monthNames.indexOf(event.target.value));
+          setDisplayMonth(new Date(displayMonth));
+        }}
+      >
+        {monthNames.map((month) => (
+          <MenuItem key={month} value={month}>
+            {month}
+          </MenuItem>
+        ))}
+      </TextField>
+
+      <TextField
+        select
+        defaultValue={displayMonth.getFullYear()}
+        onChange={(event) => {
+          displayMonth.setFullYear(Number.parseInt(event.target.value));
+          setDisplayMonth(new Date(displayMonth));
+        }}
+      >
+        {[...Array(50).keys()].map((num) => {
```

Comment:
looks like you're making a list of the next 50 years, I would do that above (like everything through `year = (num + 2024).toString()`, save it in an array then just call it down here with another map to construct the list of menuItems

Reply:
also add a comment with a todo to change this to use pagination. There's surely a better way to do this than hardcoding 50 years, but don't wanna burn time on it rn